app/views/sales/new.html.erb

```diff
@@ -0,0 +1,30 @@
+<h2>Insert a new sale</h2>
+
+<% if @sale.errors.any? %>
+  <ul class="errors">
+    <% @sale.errors.each do |column, message| %>
+      <li>
+        <strong><%= column.capitalize %></strong> <%= message %>
+      </li>
+    <% end %>
+  </ul>
+<% end %>
+
+<%= form_for @sale do |f| %>
+<!-- use the instance to give form_for all the information they need to know about tasks! -->
+  <%= field_set_tag 'Sale Information', :html => {:class => "form-control"}  do %>
+
+
+      <%= f.label :product_id %>
+      <%= f.collection_select(:product_id, Product.all, :id,  :name, :prompt => "Select Product") %>
```

Comment:
If you are assuming each vendor only sells certain products and not all products (I think the seeds are set up that way), one enhancement you could make here would be to filter the products dropdown to only include products that belong to the selected vendor. In your controller method you could create @vendor = Vendor.find(params[:id]) and then change Product.all in this line to @vendor.products because of the belongs_to/has_many relationship.